### PR TITLE
Check a rule from the rule manager

### PR DESCRIPTION
**Pull request by jonathonherbert**
## What does this change?

Adds an endpoint to check a rule manager rule from the rule manager against the specified copy.

The route is `/api/rules/:id/test-block`, and the document is specified in the body – here's an example:

```json
{
  "id": "id",
  "blocks": [
    {
      "id": "elem-0",
      "text": "The chancellor, [Jeremy Hunt](https://www.theguardian.com/politics/jeremy-hunt), said the best way to get growth going again was to ease pressure on households by bringing down inflation. “While an extra bank holiday had an impact on growth in May, high inflation remains a drag anchor on economic growth,” he said.",
      "from": 0,
      "to": 413
    }
  ]
}
```

## Dev notes

- We stream matches directly from the checker. 
- We're using HMAC auth to make the request to the checker from the rule-manager.

## How to test

- The automated tests should pass, and you should be convinced they cover the testing method. We simulate a server in the tests to test the streaming behaviour, which makes them quite comprehensive.
- Check content yourself locally! A GET request to `https://manager.typerighter.local.dev-gutools.co.uk/api/rules/:id/test-block` (be sure to add a valid cookie in the request) with the payload above should give a `200` response.

### CODE REVIEW

File: apps/rule-manager/app/service/RuleTesting.scala
Comment (jonathonherbert) [inline apps/rule-manager/app/service/RuleTesting.scala]:
```suggestion
    * service endpoint as streamed json-seq.
```

File: apps/rule-manager/test/service/RuleTestingSpec.scala
Comment (jonathonherbert) [inline apps/rule-manager/test/service/RuleTestingSpec.scala]:
```suggestion
  /** Mock responses from the checker service.
```

File: apps/checker/conf/application.conf
Comment (Fweddi) [inline apps/checker/conf/application.conf:14]: I think the url should include the prefix here - **checker**.typerighter.local.dev-gutools.co.uk.

But I can't see where this config value is used, and I'm not sure if this is how we should be using config? 

This config describes the checker app's own features, rather than configuring the checker app to use a particular property. 

It makes sense when we use it in the manager's config, but I'm not sure it makes sense here.

File: apps/rule-manager/app/controllers/RulesController.scala
Comment (Fweddi) [inline apps/rule-manager/app/controllers/RulesController.scala:201]: At one point in the commit history we handled the rule's optionality, but it seems we've lost this?

File: apps/rule-manager/app/service/RuleTesting.scala
Comment (Fweddi) [inline apps/rule-manager/app/service/RuleTesting.scala:22]: Heh, a little strange but I understand why!

Comment (Fweddi) [reply apps/rule-manager/app/service/RuleTesting.scala:22]: I guess we could add a method `draftDbRuleToCheckerRule`, but it's not a blocker (a new method might not be as DRY).

**Review by Fweddi — COMMENTED** [review]
Tested and works as expected! Nice one.

Have left a few comments - holding off on approval until CODE is green.

**Review by Fweddi — APPROVED** [review]
Tested locally and on CODE and it works as expected! Great stuff.

Have left a few non-blocking comments.

File: apps/rule-manager/app/service/RuleTesting.scala
Comment (jonathonherbert) [reply apps/rule-manager/app/service/RuleTesting.scala:22]: Yeah, it's a bit of a pain, this. Be a nice refactor down the line.

File: apps/rule-manager/app/controllers/RulesController.scala
Comment (jonathonherbert) [reply apps/rule-manager/app/controllers/RulesController.scala]: Thanks for spotting! Added in 4cde00e17460cfc26e3ad56825341ab42bf25621.

File: apps/checker/conf/application.conf
Comment (jonathonherbert) [reply apps/checker/conf/application.conf:14]: Good catch. This was likely left behind when I thought that the entire request URL was signed by HMAC. Turns out, it's just the URL's path, so this isn't necessary – removed in 0b64a29ecf9ce4c9189225214d5cf0bf02197c5c.

**Pull request merged**
